### ISSUE TITLE
dispose() causes an error with canvas renderer

### ISSUE THREAD

**Issue by dstein64** [xtermjs/xterm.js#4757]
The terminal's `dispose()` method leads to an uncaught error when using a canvas renderer. This started with the latest release (`xterm@5.3.0` and `xterm-addon-canvas@0.5.0`).

## Details
- Browser and browser version: Chrome 116
- OS version: Kubuntu 23.04
- xterm.js version: 5.3.0

### Steps to reproduce

Create a page with the following code:

```html
<!doctype html>
<html>
<head>
    <link rel="stylesheet" href="https://unpkg.com/xterm@5.3.0/css/xterm.css" />
    <script src="https://unpkg.com/xterm@5.3.0/lib/xterm.js"></script>
    <script src="https://unpkg.com/xterm-addon-canvas@0.5.0/lib/xterm-addon-canvas.js"></script>
</head>
<body>
<div id="terminal"></div>
<script>
    const term = new Terminal();
    term.loadAddon(new CanvasAddon.CanvasAddon());
    term.open(document.getElementById('terminal'));
    term.writeln('Hello world!');
</script>
</body>
</html>
```

In the developer console, type `term.dispose()`.

The following error message is printed to the console:

```
RenderService.ts:206 Uncaught TypeError: Cannot read properties of undefined (reading 'onRequestRedraw')
    at t.RenderService.setRenderer (RenderService.ts:206:26)
    at Object.dispose (CanvasAddon.ts:62:21)
    at a.dispose (Lifecycle.ts:22:9)
    at t.AddonManager._wrappedAddonDispose (AddonManager.ts:50:25)
    at t.dispose (AddonManager.ts:30:35)
    at t.AddonManager.dispose (AddonManager.ts:19:32)
    at d.dispose (Lifecycle.ts:22:9)
    at d.dispose (Terminal.ts:195:11)
    at <anonymous>:1:6
```

This does not occur when the canvas add-on is not used.

**Comment by dstein64** [comment]
`git bisect` indicates that this started at 37b0e4018cd73d3b32ab89cfcd2be62a40585944.

**Comment by Tyriar** [comment]
Thanks for looking into it, I plan on checking this out today or early next week

**Comment by Tyriar** [comment]
We can do a stable canvas addon release easily so we can fix and publish this.

**Comment by Tyriar** [comment]
@dstein64 fix is out in https://github.com/xtermjs/xterm.js/pull/4785, it's a change in core so we can't do an addon release to fix it. The change is probably harmless, you stick a try around `.dispose()` if the error is causing problems.

**Comment by BrunoQuaresma** [comment]
It is also happening with the WebGL renderer:
```console
TypeError: Cannot read properties of undefined (reading 'onRequestRedraw')
```

**Comment by catz** [comment]
+1 for Bruno issue. Reverted to 5.2.0 fixed the issue with webgl

**Comment by catz** [comment]
@Tyriar Can you please have a look it for webgl rendered ?

**Comment by Tyriar** [comment]
@catz this is fixed in 5.4.0 which hasn't been released yet. You can use the `beta` tag version in the meantime, this should be relatively safe as it's what vscode uses.

**Comment by catz** [comment]
@Tyriar Daniel, That's awesome news!
